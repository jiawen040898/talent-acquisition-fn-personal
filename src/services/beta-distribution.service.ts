export const EXPONENTIAL_SMOOTHING_WEIGHT = 0.025; // Gamma

const smooth = (alpha: number, beta: number, gamma: number): number => {
    return alpha + gamma * (beta - alpha);
};

const getGammaForUniformSmoothing = (size: number): number => {
    return 1 / size;
};

const getNewMean = (
    oldMean: number,
    value: number,
    weight = EXPONENTIAL_SMOOTHING_WEIGHT,
): number => {
    return smooth(oldMean, value, weight);
};

const getNewVariance = (
    oldVariance: number,
    oldMean: number,
    newMean: number,
    value: number,
    weight = EXPONENTIAL_SMOOTHING_WEIGHT,
): number => {
    return smooth(oldVariance, (value - oldMean) * (value - newMean), weight);
};

const getAlphaAndBeta = (
    mean: number,
    variance: number,
    size: number,
): {
    alpha: number;
    beta: number;
} => {
    if (size === 1) {
        return {
            alpha: 1,
            beta: 1,
        };
    }

    const alpha = mean * ((mean * (1 - mean)) / variance - 1);
    const beta = (1 - mean) * ((mean * (1 - mean)) / variance - 1);
    return { alpha, beta };
};

export const BetaDistributionService = {
    getGammaForUniformSmoothing,
    getNewMean,
    getNewVariance,
    getAlphaAndBeta,
};
