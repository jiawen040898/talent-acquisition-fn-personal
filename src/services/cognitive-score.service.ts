import { JobApplicationScoreType } from '@pulsifi/constants';
import { JobApplicationQuestionnaireDto } from '@pulsifi/dtos';
import { DomainTraitOutcome, logger } from '@pulsifi/fn';
import {
    AssessmentRawResult,
    CognitiveQuestionnaireFramework,
} from '@pulsifi/interfaces';
import { JobApplicationScoreMapper } from '@pulsifi/mappers';
import { JobApplicationScore } from '@pulsifi/models';
import { JobApplicationScoreUtil, QuestionnaireUtil } from '@pulsifi/shared';
import { sum } from 'lodash';
import { In } from 'typeorm';

import { getDataSource } from '../database';
import { JobApplicationScoreService } from './job-application-score.service';

const processCognitiveScore = (
    assessment: JobApplicationQuestionnaireDto,
    jobApplicationId: string,
    updatedBy: number,
    recipeId?: string,
): JobApplicationScore => {
    const scoreType =
        JobApplicationScoreMapper.mapCognitiveQuestionnaireFrameworkToScoreType[
            assessment.questionnaire_framework as CognitiveQuestionnaireFramework
        ];

    const alias =
        JobApplicationScoreMapper.mapCognitiveScoreTypeToAlias[scoreType];

    const { domain_score: domainScore, domain_alias: domainAlias } = (
        assessment?.result_raw as AssessmentRawResult
    ).scores?.find(
        (result) => result.domain_alias === alias,
    ) as DomainTraitOutcome;

    return JobApplicationScoreMapper.mapJobApplicationScore(
        JobApplicationScoreUtil.getFormattedJobApplicationScore(
            domainScore,
            scoreType,
        ),
        {
            cognitive_result: {
                domain_score: domainScore || 0,
                domain_alias: domainAlias,
                domain_percentile: null,
                ingredient_weightage: null,
            },
        },
        scoreType,
        JobApplicationScoreService.getScoreDimension(domainScore),
        jobApplicationId,
        updatedBy,
        recipeId,
    );
};

const processReasoningAverage = async (
    newScores: JobApplicationScore[],
    jobApplicationId: string,
    updatedBy: number,
    recipeId?: string,
): Promise<JobApplicationScore> => {
    const dataSource = await getDataSource();
    const existingScores = await dataSource
        .getRepository(JobApplicationScore)
        .find({
            where: {
                job_application_id: jobApplicationId,
                score_type: In([
                    JobApplicationScoreType.REASONING_LOGICAL,
                    JobApplicationScoreType.REASONING_NUMERIC,
                    JobApplicationScoreType.REASONING_VERBAL,
                ]),
            },
        });

    const newScoreTypes = newScores.map((score) => score.score_type);

    const filteredExistingScores = existingScores.filter(
        (score) => !newScoreTypes.includes(score.score_type),
    );

    const availableCognitiveScores = [...newScores, ...filteredExistingScores];

    const scores = availableCognitiveScores.map((s) =>
        JobApplicationScoreUtil.getOriginalDomainScore(
            s.score as number,
            s.score_type,
        ),
    );

    const reasoningAverageScore = sum(scores) / scores.length;

    return JobApplicationScoreMapper.mapJobApplicationScore(
        JobApplicationScoreUtil.getFormattedJobApplicationScore(
            reasoningAverageScore,
            JobApplicationScoreType.REASONING_AVG,
        ),
        {
            cognitive_result: {
                domain_score: reasoningAverageScore,
                domain_alias: JobApplicationScoreType.REASONING_AVG,
                domain_percentile: null,
                ingredient_weightage: null,
            },
        },
        JobApplicationScoreType.REASONING_AVG,
        JobApplicationScoreService.getScoreDimension(reasoningAverageScore),
        jobApplicationId,
        updatedBy,
        recipeId,
    );
};

const getCognitiveScores = async (
    assessments: JobApplicationQuestionnaireDto[],
    jobApplicationId: string,
    updatedBy: number,
    recipeId?: string,
): Promise<JobApplicationScore[]> => {
    try {
        const cognitiveScores = assessments.flatMap((assessment) => {
            if (!QuestionnaireUtil.isCognitiveQuestionnaire(assessment)) {
                return [];
            }

            return processCognitiveScore(
                assessment,
                jobApplicationId,
                updatedBy,
                recipeId,
            );
        });

        if (cognitiveScores.length === 0) {
            return [];
        }

        const reasoningAverage = await processReasoningAverage(
            cognitiveScores,
            jobApplicationId,
            updatedBy,
            recipeId,
        );

        return [...cognitiveScores, reasoningAverage];
    } catch (err) {
        logger.error('Failed to process cognitive score', {
            data: {
                assessments,
                job_application_id: jobApplicationId,
                recipe_id: recipeId,
            },
            err,
        });
        throw err;
    }
};

export const CognitiveScoreService = {
    getCognitiveScores,
    processReasoningAverage,
    processCognitiveScore,
};
