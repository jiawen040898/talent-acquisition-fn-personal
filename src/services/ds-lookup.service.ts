import { AWSConfig } from '@pulsifi/configs';
import {
    DSLookupAPIErrorCode,
    DSLookupAPIErrorMessage,
    DSLookupAPIErrorType,
} from '@pulsifi/constants';
import { ErrorDetails } from '@pulsifi/fn';
import axios from 'axios';

export interface GetPairwiseSimilarity {
    score: number;
    pairwise_score: {
        [key: string]: {
            [key: string]: number;
        };
    };
}

const getPairwiseSimilarity = async (
    main: JSON | string[] | null,
    secondary: string[] | null,
): Promise<GetPairwiseSimilarity | undefined> => {
    const payload = {
        main,
        secondary,
    };

    const path = `${AWSConfig().alb.dns}/lookup/ds/pairwise-similarity/score`;

    try {
        const res = await axios.post<GetPairwiseSimilarity>(path, payload);

        return res.data;
    } catch (err) {
        throw new FailedToGetPairwiseSimilarityFromDSLookupApi({
            error_codes: [
                DSLookupAPIErrorCode.FAILED_TO_GET_PAIRWISE_SIMILARITY,
            ],
            err,
        });
    }
};

export class FailedToGetPairwiseSimilarityFromDSLookupApi extends Error {
    errorDetails: ErrorDetails;

    constructor(errorDetails: ErrorDetails) {
        super(DSLookupAPIErrorMessage.FAILED_TO_GET_PAIRWISE_SIMILARITY);
        this.name = DSLookupAPIErrorType.FAILED_TO_GET_PAIRWISE_SIMILARITY;
        this.errorDetails = errorDetails;
    }
}

export const DSLookupService = {
    getPairwiseSimilarity,
};
