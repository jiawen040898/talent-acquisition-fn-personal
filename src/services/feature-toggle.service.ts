import { UnleashConfig } from '@pulsifi/configs';
import {
    UnleashErrorCode,
    UnleashErrorMessage,
    UnleashErrorType,
} from '@pulsifi/constants';
import { ErrorDetails } from '@pulsifi/fn';
import { startUnleash, Unleash } from 'unleash-client';

let unleash: Unleash | null = null;

const initUnleash = () => {
    const config = UnleashConfig();
    const url = config.apiUrl;
    const projectId = config.projectId;
    const authToken = config.apiKey;

    return startUnleash({
        url: url,
        appName: projectId,
        customHeaders: { Authorization: authToken },
    });
};

const isUnleashFlagEnabled = async (
    flag: SafeAny,
    unleashContext: { [key: string]: string },
): Promise<boolean> => {
    try {
        if (!unleash) {
            unleash = await initUnleash();
        }

        return unleash.isEnabled(flag, unleashContext);
    } catch (error) {
        throw new FailedToGetUnleashEnabledFlag({
            error_codes: [UnleashErrorCode.FAILED_TO_GET_UNLEASH_ENABLED_FLAG],
            error,
        });
    }
};

export class FailedToGetUnleashEnabledFlag extends Error {
    errorDetails: ErrorDetails;

    constructor(errorDetails: ErrorDetails) {
        super(UnleashErrorMessage.FAILED_TO_GET_UNLEASH_ENABLED_FLAG);
        this.name = UnleashErrorType.FAILED_TO_GET_UNLEASH_ENABLED_FLAG;
        this.errorDetails = errorDetails;
    }
}

export const FeatureToggleService = {
    isUnleashFlagEnabled,
};
