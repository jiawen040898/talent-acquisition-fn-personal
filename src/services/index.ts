export * from './beta-distribution.service';
export * from './cognitive-score.service';
export * from './ds-lookup.service';
export * from './feature-toggle.service';
export * from './fit-score.service';
export * from './job-application.service';
export * from './job-application-score.service';
export * from './job-application-skill-extraction.service';
export * from './job-distribution-score.service';
export * from './job-user-access.service';
export * from './pairwise-score.service';
export * from './personality-score.service';
export * from './psychology.service';
export * from './publisher.service';
export * from './s3.service';
export * from './skill-extract-api.service';
