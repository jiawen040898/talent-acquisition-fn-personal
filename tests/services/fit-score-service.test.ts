import { FitScoreService } from '@pulsifi/services';
import {
    mockCultureFitRecipe,
    mockRoleFitRecipe,
    mockRoleFitRecipeWithCompetencyInclusiveness,
} from '@pulsifi/tests/fixtures';
import {
    allCompletedNonFitScores,
    reasoningLogicalScore,
    reasoningNumericalScore,
    reasoningVerbalScore,
    workInterestScore,
    workStyleScore,
    workValuesScore,
} from '@pulsifi/tests/fixtures/job-application-score/test-data';

const jobApplicationId = '00000000-0000-0000-0000-000000000001';

describe('FitScoreService', () => {
    describe('mapRoleFitDomainScores', () => {
        it('should return correct output', () => {
            // Act
            const result = FitScoreService.mapRoleFitDomainScores(
                allCompletedNonFitScores,
            );

            // Assert
            expect(result).toMatchSnapshot();
        });
    });

    describe('getRoleFitJobApplicationScore', () => {
        it('should return correct output', () => {
            // Act
            const result = FitScoreService.getRoleFitJobApplicationScore(
                allCompletedNonFitScores,
                mockRoleFitRecipe,
                mockRoleFitRecipe.id,
                jobApplicationId,
                5,
            );

            // Assert
            expect(result).toMatchSnapshot();
        });

        it('should return void if not all required scores are present for role fit score calculation', () => {
            // Act
            const result = FitScoreService.getRoleFitJobApplicationScore(
                [workStyleScore, workValuesScore],
                mockRoleFitRecipe,
                mockRoleFitRecipe.id,
                jobApplicationId,
                5,
            );

            // Assert
            expect(result).toBeUndefined();
        });
    });

    describe('mapCultureFitDomainScore', () => {
        it('should return correct output', () => {
            // Act
            const result = FitScoreService.mapCultureFitDomainScore(
                allCompletedNonFitScores,
            );

            // Assert
            expect(result).toMatchSnapshot();
        });
    });

    describe('getCultureFitJobApplicationScore', () => {
        it('should return correct output', () => {
            // Act
            const result = FitScoreService.getCultureFitJobApplicationScore(
                allCompletedNonFitScores,
                mockCultureFitRecipe,
                mockCultureFitRecipe.id,
                jobApplicationId,
                5,
            );

            // Assert
            expect(result).toMatchSnapshot();
        });

        it('should return void if not all required scores are present for culture fit score calculation', () => {
            // Act
            const result = FitScoreService.getCultureFitJobApplicationScore(
                [workStyleScore, workValuesScore],
                mockCultureFitRecipe,
                mockCultureFitRecipe.id,
                jobApplicationId,
                5,
            );

            // Assert
            expect(result).toBeUndefined();
        });
    });

    describe('hasAllRequiredScoresForCalculation', () => {
        it.each([
            {
                scores: [],
                recipe: mockCultureFitRecipe,
                result: false,
                name: 'empty scores',
                recipeName: 'culture fit recipe',
            },
            {
                scores: [],
                recipe: mockRoleFitRecipe,
                result: false,
                name: 'empty scores',
                recipeName: 'role fit recipe',
            },
            {
                scores: allCompletedNonFitScores,
                recipe: mockCultureFitRecipe,
                result: true,
                name: 'all completed non fit scores',
                recipeName: 'culture fit recipe',
            },
            {
                scores: allCompletedNonFitScores,
                recipe: mockRoleFitRecipe,
                result: true,
                name: 'all completed non fit scores',
                recipeName: 'role fit recipe',
            },
            {
                scores: allCompletedNonFitScores,
                recipe: mockRoleFitRecipeWithCompetencyInclusiveness,
                result: true,
                name: 'all completed non fit scores',
                recipeName: 'role fit recipe with competency inclusiveness',
            },
            {
                scores: [workStyleScore, workValuesScore, workInterestScore],
                recipe: mockCultureFitRecipe,
                result: true,
                name: 'work style, work values, work interest scores',
                recipeName: 'culture fit recipe',
            },
            {
                scores: [workStyleScore, workValuesScore, workInterestScore],
                recipe: mockRoleFitRecipe,
                result: false,
                name: 'work style, work values, work interest scores',
                recipeName: 'role fit recipe',
            },
            {
                scores: [workStyleScore, workValuesScore],
                recipe: mockCultureFitRecipe,
                result: false,
                name: 'work style, work values scores',
                recipeName: 'culture fit recipe',
            },
            {
                scores: [workStyleScore, workValuesScore, workInterestScore],
                recipe: mockRoleFitRecipe,
                result: false,
                name: 'work style, work values, work interest scores',
                recipeName: 'role fit recipe',
            },
            {
                scores: [
                    workStyleScore,
                    workValuesScore,
                    workInterestScore,
                    workInterestScore,
                    reasoningLogicalScore,
                    reasoningVerbalScore,
                    reasoningNumericalScore,
                ],
                recipe: mockRoleFitRecipe,
                result: true,
                name: 'all score except pairwise skill',
                recipeName: 'role fit recipe',
            },
            {
                scores: [
                    workStyleScore,
                    workValuesScore,
                    workInterestScore,
                    reasoningLogicalScore,
                    reasoningVerbalScore,
                    reasoningNumericalScore,
                ],
                recipe: mockRoleFitRecipeWithCompetencyInclusiveness,
                result: false,
                name: 'all score except pairwise skill',
                recipeName: 'role fit recipe with competency inclusiveness',
            },
        ])(
            `should return $result when score given is $name and recipe is $recipeName`,
            ({ scores: jobApplicationScores, recipe, result }) => {
                // Act
                const isCalculated =
                    FitScoreService.hasAllRequiredScoresForCalculation(
                        recipe,
                        jobApplicationScores,
                    );

                // Assert
                expect(isCalculated).toEqual(result);
            },
        );
    });
});
