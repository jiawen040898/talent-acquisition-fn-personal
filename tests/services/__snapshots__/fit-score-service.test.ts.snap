// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FitScoreService getCultureFitJobApplicationScore should return correct output 1`] = `
{
  "created_by": 5,
  "job_application_id": "00000000-0000-0000-0000-000000000001",
  "score": 6.1308688,
  "score_dimension": 1,
  "score_outcome": {
    "framework_alias": "polaris",
    "personality_result": [
      {
        "domain_alias": "people_centric",
        "domain_score": 0.5587499,
        "domain_weightage": 0.125,
        "traits": [
          {
            "trait_alias": "social",
            "trait_framework": "work_interest",
            "trait_score": 0.625,
            "trait_weightage": 0.2,
          },
          {
            "trait_alias": "support",
            "trait_framework": "work_value",
            "trait_score": 0.49166666666666664,
            "trait_weightage": 0.2,
          },
          {
            "trait_alias": "cooperation",
            "trait_framework": "work_style",
            "trait_score": 0.5,
            "trait_weightage": 0.2,
          },
          {
            "trait_alias": "concern_for_others",
            "trait_framework": "work_style",
            "trait_score": 0.6666666666666667,
            "trait_weightage": 0.2,
          },
          {
            "trait_alias": "social_orientation",
            "trait_framework": "work_style",
            "trait_score": 0.5104166666666666,
            "trait_weightage": 0.2,
          },
        ],
      },
      {
        "domain_alias": "ownership",
        "domain_score": 0.5581786,
        "domain_weightage": 0.125,
        "traits": [
          {
            "trait_alias": "investigative",
            "trait_framework": "work_interest",
            "trait_score": 0.71875,
            "trait_weightage": 0.2,
          },
          {
            "trait_alias": "independence",
            "trait_framework": "work_value",
            "trait_score": 0.4488095238095238,
            "trait_weightage": 0.2,
          },
          {
            "trait_alias": "relationships",
            "trait_framework": "work_value",
            "trait_score": 0.49833333333333335,
            "trait_weightage": 0.2,
          },
          {
            "trait_alias": "leadership",
            "trait_framework": "work_style",
            "trait_score": 0.5416666666666666,
            "trait_weightage": 0.2,
          },
          {
            "trait_alias": "integrity",
            "trait_framework": "work_style",
            "trait_score": 0.5833333333333334,
            "trait_weightage": 0.2,
          },
        ],
      },
      {
        "domain_alias": "learning",
        "domain_score": 0.6639881000000001,
        "domain_weightage": 0.125,
        "traits": [
          {
            "trait_alias": "working_conditions",
            "trait_framework": "work_value",
            "trait_score": 0.49523809523809526,
            "trait_weightage": 0.25,
          },
          {
            "trait_alias": "achievement_effort",
            "trait_framework": "work_style",
            "trait_score": 0.8229166666666666,
            "trait_weightage": 0.25,
          },
          {
            "trait_alias": "initiative",
            "trait_framework": "work_style",
            "trait_score": 0.7232142857142857,
            "trait_weightage": 0.25,
          },
          {
            "trait_alias": "independence",
            "trait_framework": "work_style",
            "trait_score": 0.6145833333333334,
            "trait_weightage": 0.25,
          },
        ],
      },
      {
        "domain_alias": "avant_garde",
        "domain_score": 0.6029731,
        "domain_weightage": 0.125,
        "traits": [
          {
            "trait_alias": "realistic",
            "trait_framework": "work_interest",
            "trait_score": 0.71875,
            "trait_weightage": 0.1428571,
          },
          {
            "trait_alias": "reasoning_logical",
            "trait_framework": "pulsifi_default",
            "trait_score": 0.77832782,
            "trait_weightage": 0.1428571,
          },
          {
            "trait_alias": "independence",
            "trait_framework": "work_value",
            "trait_score": 0.4488095238095238,
            "trait_weightage": 0.1428571,
          },
          {
            "trait_alias": "working_conditions",
            "trait_framework": "work_value",
            "trait_score": 0.49523809523809526,
            "trait_weightage": 0.1428571,
          },
          {
            "trait_alias": "adaptability_flexibility",
            "trait_framework": "work_style",
            "trait_score": 0.6796875,
            "trait_weightage": 0.1428571,
          },
          {
            "trait_alias": "innovation",
            "trait_framework": "work_style",
            "trait_score": 0.6625,
            "trait_weightage": 0.1428571,
          },
          {
            "trait_alias": "analytical_thinking",
            "trait_framework": "work_style",
            "trait_score": 0.4375,
            "trait_weightage": 0.1428571,
          },
        ],
      },
      {
        "domain_alias": "results_driven",
        "domain_score": 0.6051785999999999,
        "domain_weightage": 0.125,
        "traits": [
          {
            "trait_alias": "achievement",
            "trait_framework": "work_value",
            "trait_score": 0.6071428571428571,
            "trait_weightage": 0.2,
          },
          {
            "trait_alias": "recognition",
            "trait_framework": "work_value",
            "trait_score": 0.4083333333333334,
            "trait_weightage": 0.2,
          },
          {
            "trait_alias": "achievement_effort",
            "trait_framework": "work_style",
            "trait_score": 0.8229166666666666,
            "trait_weightage": 0.2,
          },
          {
            "trait_alias": "persistence",
            "trait_framework": "work_style",
            "trait_score": 0.5625,
            "trait_weightage": 0.2,
          },
          {
            "trait_alias": "dependability",
            "trait_framework": "work_style",
            "trait_score": 0.625,
            "trait_weightage": 0.2,
          },
        ],
      },
      {
        "domain_alias": "in_sync",
        "domain_score": 0.5739584,
        "domain_weightage": 0.125,
        "traits": [
          {
            "trait_alias": "support",
            "trait_framework": "work_value",
            "trait_score": 0.49166666666666664,
            "trait_weightage": 0.25,
          },
          {
            "trait_alias": "concern_for_others",
            "trait_framework": "work_style",
            "trait_score": 0.6666666666666667,
            "trait_weightage": 0.25,
          },
          {
            "trait_alias": "self_control",
            "trait_framework": "work_style",
            "trait_score": 0.5125,
            "trait_weightage": 0.25,
          },
          {
            "trait_alias": "stress_tolerance",
            "trait_framework": "work_style",
            "trait_score": 0.625,
            "trait_weightage": 0.25,
          },
        ],
      },
      {
        "domain_alias": "stakeholder_savvy",
        "domain_score": 0.5563333,
        "domain_weightage": 0.125,
        "traits": [
          {
            "trait_alias": "enterprising",
            "trait_framework": "work_interest",
            "trait_score": 0.71875,
            "trait_weightage": 0.2,
          },
          {
            "trait_alias": "relationships",
            "trait_framework": "work_value",
            "trait_score": 0.49833333333333335,
            "trait_weightage": 0.2,
          },
          {
            "trait_alias": "leadership",
            "trait_framework": "work_style",
            "trait_score": 0.5416666666666666,
            "trait_weightage": 0.2,
          },
          {
            "trait_alias": "social_orientation",
            "trait_framework": "work_style",
            "trait_score": 0.5104166666666666,
            "trait_weightage": 0.2,
          },
          {
            "trait_alias": "self_control",
            "trait_framework": "work_style",
            "trait_score": 0.5125,
            "trait_weightage": 0.2,
          },
        ],
      },
      {
        "domain_alias": "plus",
        "domain_score": 0.785335,
        "domain_weightage": 0.125,
        "traits": [
          {
            "trait_alias": "reasoning_verbal",
            "trait_framework": "pulsifi_default",
            "trait_score": 0.73232432,
            "trait_weightage": 0.33333,
          },
          {
            "trait_alias": "reasoning_logical",
            "trait_framework": "pulsifi_default",
            "trait_score": 0.77832782,
            "trait_weightage": 0.33333,
          },
          {
            "trait_alias": "reasoning_numeric",
            "trait_framework": "pulsifi_default",
            "trait_score": 0.8453762684,
            "trait_weightage": 0.33333,
          },
        ],
      },
    ],
  },
  "score_recipe_id": "a806101e-3710-4d79-b854-bf3820c9d585",
  "score_type": "culture_fit",
  "updated_by": 5,
}
`;

exports[`FitScoreService getRoleFitJobApplicationScore should return correct output 1`] = `
{
  "created_by": 5,
  "job_application_id": "00000000-0000-0000-0000-000000000001",
  "score": 6.195649,
  "score_dimension": 1,
  "score_outcome": {
    "ingredient_result": [
      {
        "display_score_rounding_adjustment": 0.01,
        "display_weightage_rounding_adjustment": 0,
        "ingredient_alias": "reasoning_logical",
        "ingredient_score": 0.7783278199999999,
        "ingredient_weightage": 0.083333333,
      },
      {
        "display_score_rounding_adjustment": 0,
        "display_weightage_rounding_adjustment": 0,
        "ingredient_alias": "interest_riasec",
        "ingredient_score": 0.57143,
        "ingredient_weightage": 0.25,
      },
      {
        "display_score_rounding_adjustment": 0,
        "display_weightage_rounding_adjustment": 0.01,
        "ingredient_alias": "reasoning_verbal",
        "ingredient_score": 0.73232432,
        "ingredient_weightage": 0.083333333,
      },
      {
        "display_score_rounding_adjustment": 0,
        "display_weightage_rounding_adjustment": 0,
        "ingredient_alias": "reasoning_numeric",
        "ingredient_score": 0.8453762684,
        "ingredient_weightage": 0.083333333,
      },
      {
        "display_score_rounding_adjustment": 0,
        "display_weightage_rounding_adjustment": 0,
        "ingredient_alias": "work_style",
        "ingredient_score": 0.6010443000000001,
        "ingredient_weightage": 0.25,
      },
      {
        "display_score_rounding_adjustment": 0,
        "display_weightage_rounding_adjustment": 0,
        "ingredient_alias": "work_value",
        "ingredient_score": 0.5204422,
        "ingredient_weightage": 0.25,
      },
    ],
  },
  "score_recipe_id": "106e0b1d-02c4-4321-9dac-9768f84f8bc1",
  "score_type": "role_fit",
  "updated_by": 5,
}
`;

exports[`FitScoreService mapCultureFitDomainScore should return correct output 1`] = `
[
  {
    "domain_alias": "achievement_effort",
    "domain_framework": "work_style",
    "domain_score": 0.8229166666666666,
  },
  {
    "domain_alias": "persistence",
    "domain_framework": "work_style",
    "domain_score": 0.5625,
  },
  {
    "domain_alias": "initiative",
    "domain_framework": "work_style",
    "domain_score": 0.7232142857142857,
  },
  {
    "domain_alias": "leadership",
    "domain_framework": "work_style",
    "domain_score": 0.5416666666666666,
  },
  {
    "domain_alias": "cooperation",
    "domain_framework": "work_style",
    "domain_score": 0.5,
  },
  {
    "domain_alias": "concern_for_others",
    "domain_framework": "work_style",
    "domain_score": 0.6666666666666667,
  },
  {
    "domain_alias": "social_orientation",
    "domain_framework": "work_style",
    "domain_score": 0.5104166666666666,
  },
  {
    "domain_alias": "self_control",
    "domain_framework": "work_style",
    "domain_score": 0.5125,
  },
  {
    "domain_alias": "stress_tolerance",
    "domain_framework": "work_style",
    "domain_score": 0.625,
  },
  {
    "domain_alias": "adaptability_flexibility",
    "domain_framework": "work_style",
    "domain_score": 0.6796875,
  },
  {
    "domain_alias": "dependability",
    "domain_framework": "work_style",
    "domain_score": 0.625,
  },
  {
    "domain_alias": "attention_to_detail",
    "domain_framework": "work_style",
    "domain_score": 0.4375,
  },
  {
    "domain_alias": "integrity",
    "domain_framework": "work_style",
    "domain_score": 0.5833333333333334,
  },
  {
    "domain_alias": "independence",
    "domain_framework": "work_style",
    "domain_score": 0.6145833333333334,
  },
  {
    "domain_alias": "innovation",
    "domain_framework": "work_style",
    "domain_score": 0.6625,
  },
  {
    "domain_alias": "analytical_thinking",
    "domain_framework": "work_style",
    "domain_score": 0.4375,
  },
  {
    "domain_alias": "achievement",
    "domain_framework": "work_value",
    "domain_score": 0.6071428571428571,
  },
  {
    "domain_alias": "working_conditions",
    "domain_framework": "work_value",
    "domain_score": 0.49523809523809526,
  },
  {
    "domain_alias": "recognition",
    "domain_framework": "work_value",
    "domain_score": 0.4083333333333334,
  },
  {
    "domain_alias": "relationships",
    "domain_framework": "work_value",
    "domain_score": 0.49833333333333335,
  },
  {
    "domain_alias": "support",
    "domain_framework": "work_value",
    "domain_score": 0.49166666666666664,
  },
  {
    "domain_alias": "independence",
    "domain_framework": "work_value",
    "domain_score": 0.4488095238095238,
  },
  {
    "domain_alias": "realistic",
    "domain_framework": "work_interest",
    "domain_score": 0.71875,
  },
  {
    "domain_alias": "investigative",
    "domain_framework": "work_interest",
    "domain_score": 0.71875,
  },
  {
    "domain_alias": "artistic",
    "domain_framework": "work_interest",
    "domain_score": 0.46875,
  },
  {
    "domain_alias": "social",
    "domain_framework": "work_interest",
    "domain_score": 0.625,
  },
  {
    "domain_alias": "enterprising",
    "domain_framework": "work_interest",
    "domain_score": 0.71875,
  },
  {
    "domain_alias": "conventional",
    "domain_framework": "work_interest",
    "domain_score": 0.53125,
  },
  {
    "domain_alias": "reasoning_verbal",
    "domain_score": 0.73232432,
  },
  {
    "domain_alias": "reasoning_logical",
    "domain_score": 0.77832782,
  },
  {
    "domain_alias": "reasoning_numeric",
    "domain_score": 0.8453762684,
  },
]
`;

exports[`FitScoreService mapRoleFitDomainScores should return correct output 1`] = `
[
  {
    "domain_alias": "work_style",
    "domain_score": 0.6010443000000001,
  },
  {
    "domain_alias": "work_value",
    "domain_score": 0.5204422,
  },
  {
    "domain_alias": "interest_riasec",
    "domain_score": 0.57143,
  },
  {
    "domain_alias": "work_experience",
    "domain_score": 0.8,
  },
  {
    "domain_alias": "reasoning_verbal",
    "domain_score": 0.73232432,
  },
  {
    "domain_alias": "reasoning_logical",
    "domain_score": 0.7783278199999999,
  },
  {
    "domain_alias": "reasoning_numeric",
    "domain_score": 0.8453762684,
  },
  {
    "domain_alias": "hard_skills",
    "domain_score": 0.8,
  },
]
`;
